internal/services/dynatrace/dynatrace_monitors_data_source.go

```diff
@@ -0,0 +1,217 @@
+	EnvironmentProperties         []EnvironmentProperties        `tfschema:"environment_properties"`
+	PlanData                      []PlanData                     `tfschema:"plan"`
+	UserInfo                      []UserInfo                     `tfschema:"user"`
+	Tags                          map[string]string              `tfschema:"tags"`
+}
+
+type EnvironmentProperties struct {
+	EnvironmentInfo []EnvironmentInfo `tfschema:"environment_info"`
+}
+
+type EnvironmentInfo struct {
+	EnvironmentId string `tfschema:"environment_id"`
+}
+
+var _ sdk.DataSource = MonitorsDataSource{}
+
+func (d MonitorsDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+	}
+}
+
+func (d MonitorsDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationComputed(),
+
+		"monitoring_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"marketplace_subscription": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"identity": commonschema.SystemAssignedIdentityComputed(),
+
+		"plan": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"billing_cycle": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"plan": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"usage_type": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"effective_date": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"user": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"country": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"email": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"first_name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"last_name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"phone_number": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"environment_properties": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"environment_info": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"environment_id": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
Is there a reason we're doing 2 levels of nesting here for a single value? Can this just be flattened to the top level?

Also, this property doesn't appear to be supported by the related resource, should it even be included here?